### PR TITLE
Use name prefix instead of service UUID

### PR DESCRIPTION
**Pull request by g-ortuno**
Some devices advertise a different service UUID.

### CODE REVIEW

**Comment by g-ortuno** [comment]
@entertailion do you think this is OK?

File: bluetooth-toy-bb8/index.html [hunk 1]
```diff
@@ -264,7 +264,7 @@
           if (controlCharacteristic == null) {
             navigator.bluetooth.requestDevice({
```
Comment (jyasskin) [inline bluetooth-toy-bb8/index.html:265]: Be sure to either request '22bb746f-2bb0-7554-2d6f-726568705327' as an optional service, or include it in a second filter.

**Comment by jyasskin** [comment]
Does the rest of the code check for any services it needs after it connects? With just a name prefix it's more likely to receive a bad device, and need to tell the user to try again.

**Comment by LeonNicholls** [comment]
I will need to do some testing to understand the impact of this change.

File: bluetooth-toy-bb8/index.html (hunk 1 above)
Comment (LeonNicholls) [inline bluetooth-toy-bb8/index.html:265]: Just got time to look at this. Agree with @jyasskin. The name prefix is too generic. Not accepting this pull request.

**Comment by g-ortuno** [comment]
The problem is that there are BB-8s out there that are advertising a different UUID. What should we do for these BB-8s? Are these just not supported?

**Comment by LeonNicholls** [comment]
Have you tried using multiple services in the filter:
navigator.bluetooth.requestDevice({
  filters: [
    {services: [A]},
    {services: [B]}
  ]
});

**Comment by beaufortfrancois** [comment]
Do they only advertise a different service UUID but use the same one when discovered?
If not, having a name Prefix won't help.

**Comment by g-ortuno** [comment]
Looked up the services with the nordic app. Yes, our BB-8 still contains the old services and everything works fine after connecting.

**Comment by jyasskin** [comment]
If the advertised service is still distinctive, adding a filter for that one should work well. If it's a widely-used one (like a battery service), then filtering for it is no better than the short name prefix.

It'd still be good to have a path in the code to handle a device that doesn't contain the expected service.

**Pull request closed without merging**